certora/specs/SocialRecoveryModule.spec

```diff
@@ -502,3 +503,113 @@ rule finalizeRecoveryAlwaysPossible(env e) {
 
     assert !isReverted, "legitimate recovery finalization reverted";
 }
+
+// This rule verifies that if recovery request data is changed, it must be one of the following functions:
+// - confirmRecovery(...)
+// - multiConfirmRecovery(...)
+// - executeRecovery(...)
+// - finalizeRecovery(...)
+// - cancelRecovery(...)
+// Each of these either updates or deletes the recovery request.
+rule recoveryRequestsChange(env e, calldataarg args, method f) {
+    uint i;
+    uint256 guardianApprovalCountBefore = currentContract.recoveryRequests[safeContract].guardiansApprovalCount;
+    uint256 newThresholdBefore = currentContract.recoveryRequests[safeContract].newThreshold;
+    uint64 executeAfterBefore = currentContract.recoveryRequests[safeContract].executeAfter;
+    address newOwnersBefore = currentContract.recoveryRequests[safeContract].newOwners[i];
+
+    f(e, args);
+
+    uint256 guardianApprovalCountAfter = currentContract.recoveryRequests[safeContract].guardiansApprovalCount;
+    uint256 newThresholdAfter = currentContract.recoveryRequests[safeContract].newThreshold;
+    uint64 executeAfterAfter = currentContract.recoveryRequests[safeContract].executeAfter;
+    address newOwnersAfter = currentContract.recoveryRequests[safeContract].newOwners[i];
+
+    assert (
+        guardianApprovalCountBefore != guardianApprovalCountAfter ||
+        newThresholdBefore != newThresholdAfter ||
+        executeAfterBefore != executeAfterAfter ||
+        newOwnersBefore != newOwnersAfter
+        ) =>
+        f.selector == sig:confirmRecovery(address,address[],uint256,bool).selector ||
+        f.selector == sig:multiConfirmRecovery(address,address[],uint256,SocialRecoveryModule.SignatureData[],bool).selector ||
+        f.selector == sig:executeRecovery(address,address[],uint256).selector ||
+        f.selector == sig:finalizeRecovery(address).selector ||
+        f.selector == sig:cancelRecovery().selector;
+}
+
+// This rule verifies that is the confirmedHashes change, it must be one of the following functions:
+// - confirmRecovery(...)
+// - multiConfirmRecovery(...)
+rule confirmedHashesChange(env e, calldataarg args, method f, bytes32 hash, address guardian) {
```

Comment:
Same comment here about the env and calldataarg. Note that I like the hash and guardian here, as they are related to what cannot change.

Reply:
Done 👍🏾